app/src/main/java/com/mashup/mobalmobal/ui/profile/presenter/ProfileAdapter.kt

```diff
@@ -0,0 +1,107 @@
+            is ProfileItem.RequestDonation -> ViewType.REQUEST_DONATION.ordinal
+            is ProfileItem.Donated -> ViewType.DONATED.ordinal
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            ViewType.USER.ordinal -> ProfileUserViewHolder(
+                HolderProfileUserBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                ), clickListener
+            )
+            ViewType.POINT.ordinal -> ProfilePointViewHolder(
+                HolderProfilePointBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                ), clickListener
+            )
+            ViewType.DONATION_SUMMARY.ordinal -> ProfileDonationSummaryViewHolder(
+                HolderProfileDonationSummaryBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                ), clickListener
+            )
+            ViewType.REQUEST_DONATION.ordinal -> ProfileRequestDonationViewHolder(
+                HolderProfileRequestDonationBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                ), clickListener
+            )
+            ViewType.DONATED.ordinal -> ProfileDonatedViewHolder(
+                HolderProfileDonatedBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                ), clickListener
+            )
+            else -> throw IllegalArgumentException()
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is ProfileUserViewHolder -> {
+                holder.bind(getItem(position) as ProfileItem.User)
+            }
+            is ProfilePointViewHolder -> {
+                holder.bind(getItem(position) as ProfileItem.Point)
+            }
+            is ProfileDonationSummaryViewHolder -> {
+                holder.bind(getItem(position) as ProfileItem.DonationSummary)
+            }
+            is ProfileRequestDonationViewHolder -> {
+                holder.bind(getItem(position) as ProfileItem.RequestDonation)
+            }
+            is ProfileDonatedViewHolder -> {
+                holder.bind(getItem(position) as ProfileItem.Donated)
+            }
+        }
+    }
+
+    enum class ViewType {
+        USER,
+        POINT,
+        DONATION_SUMMARY,
+        REQUEST_DONATION,
+        DONATED
+    }
+
+    class ProfileItemDiffCallback : DiffUtil.ItemCallback<ProfileItem>() {
+        override fun areItemsTheSame(oldItem: ProfileItem, newItem: ProfileItem): Boolean {
+            return oldItem == newItem
+        }
```

Comment:
이러면 areContentsTheSame과 다를바가 없어요!
ProfileItem에 id를 추가하면 좋을 것 같아요.
```suggestion
        override fun areItemsTheSame(oldItem: ProfileItem, newItem: ProfileItem): Boolean {
            return oldItem.id == newItem.id
        }
```

Reply:
수정했습니다~